src/Components/ADT3DSceneBuilder/Internal/Behaviors/Internal/VisualRulesTab.tsx

```diff
@@ -17,50 +36,94 @@ const LOC_KEYS = {
-     * UPDATES RULE WILL NEED TO PROPAGATE TO BEHAVIOR STATE
-     *  */
-    const mockData: IVisualRule[] = [
-        {
-            id: 'Machine temp',
-            displayName: 'Machine temp',
-            conditions: ['Badge', 'Mesh coloring']
-        },
-        {
-            id: 'Machine health',
-            displayName: 'Machine health',
-            conditions: ['Mesh coloring']
+    //hooks
+    const { t } = useTranslation();
+
+    const {
+        behaviorFormState,
+        behaviorFormDispatch
+    } = useBehaviorFormContext();
+    const { visualRuleFormMode, setVisualRuleFormMode } = useContext(
+        SceneBuilderContext
+    );
+
+    //state
+    const [ruleItems, setRuleItems] = useState<IVisualRule[]>([]);
+    const [
+        isPropertyTypeDropdownEnabled,
+        setisPropertyTypeDropdownEnabled
+    ] = useState(false);
+
+    const isVisualRulesFormActive =
+        visualRuleFormMode !== VisualRuleFormMode.Inactive;
+    const formHeaderHeight = isVisualRulesFormActive
+        ? isPropertyTypeDropdownEnabled
+            ? 370
+            : 268
+        : 168;
+
+    //callbacks
+    const onRemoveRule = useCallback(
+        (id: string) => {
+            behaviorFormDispatch({
+                type:
+                    BehaviorFormContextActionType.FORM_BEHAVIOR_VISUAL_RULE_REMOVE,
+                payload: {
+                    visualRuleId: id
+                }
+            });
         },
-        {
-            id: 'Machine active',
-            displayName: 'Machine active',
-            conditions: ['Mesh coloring']
+        [behaviorFormDispatch]
+    );
+
+    const onEditRule = useCallback(() => {
+        setVisualRuleFormMode(VisualRuleFormMode.EditVisualRule);
+    }, []);
+
+    //side effects
+
+    //get all the Visual Rules from behavior
+    const visualRules: IExpressionRangeVisual[] = useMemo(() => {
+        return (
+            getVisualRulesFromBehavior(behaviorFormState.behaviorToEdit) || []
+        );
+    }, [behaviorFormState.behaviorToEdit]);
+
+    //transform from elements from IExpressionVisual => IVisualRule
+    const getVisualList = () => {
+        let rules: IVisualRule[] = [];
+        if (visualRules.length > 0) {
+            rules = visualRules.map((rule) => {
+                return {
+                    id: rule.id,
+                    displayName: rule?.displayName
+                        ? rule.displayName
+                        : '(unlabeled)',
+                    conditions: rule.valueRanges,
+                    type: rule.expressionType
+                };
+            });
         }
-    ];
-    const [ruleItems] = useState<IVisualRule[]>(mockData);
-    const [
-        isVisualRuleFormOpen,
-        { toggle: toggleIsVisualRuleFormOpen }
-    ] = useBoolean(false);
+        return rules;
+    };
+
+    // set the list IVisualRules that are going to be shown
+    useEffect(() => {
+        setRuleItems(getVisualList());
```

Comment:
since this has the behaviorFormState in the dependency list, can we pass it into the function as an argument so that the dependency is actually used? It is a bit suspect when we have a depencency on something that is not actually used in the `useEffect` directly. That means that we have to have some knowledge of how that function works internally which smells bad to me.